packages/Php70/src/Rector/FuncCall/NonVariableToVariableOnFunctionCallRector.php

```diff
@@ -0,0 +1,204 @@
+                    $variable = new Variable(lcfirst($argument->class->name->toString()));
+                } else {
+                    continue;
+                }
+
+                $assign = new Node\Expr\Assign($variable, $argument);
+            }
+
+            if ($variable === null || $assign === null) {
+                continue;
+            }
+
+            $this->addNodeBeforeNode($assign, $node);
+            $node->args[$key]->value = $variable;
+        }
+
+        return $node;
+    }
+
+    /**
+     * @param FuncCall|Node\Expr\MethodCall|Node\Expr\StaticCall $node
+     *
+     * @return array<int, Node\Expr>
+     */
+    private function getNonVariableArguments($node): array
+    {
+        $arguments = [];
+
+        foreach ($this->getParameters($node) as $key => $parameter) {
+            if ($parameter->passedByReference()->no()) {
+                continue;
+            }
+
+            $argument = $node->args[$key]->value;
+
+            if (
+                $argument instanceof Variable
+                || $argument instanceof ArrayDimFetch
+                || $argument instanceof PropertyFetch
+                || $argument instanceof StaticPropertyFetch
+            ) {
+                continue;
+            }
+
+            $arguments[$key] = $argument;
+        }
+
+        return $arguments;
+    }
+
+    /**
+     * @param FuncCall|Node\Expr\MethodCall|Node\Expr\StaticCall $node
+     *
+     * @return array<int, ParameterReflection>
+     */
+    private function getParameters($node): array
+    {
+        if ($node->name instanceof Identifier) {
+            $classType = $this->getObjectType($node);
+            $methodName = $this->getName($node->name);
+
+            if ($methodName === null || ! $classType->hasMethod($methodName)->yes()) {
+                return [];
+            }
+
+            return ParametersAcceptorSelector::selectSingle(
+                $classType->getMethod($methodName, $node->getAttribute(AttributeKey::SCOPE))->getVariants()
+            )->getParameters();
+        }
+
+        if ($node->name instanceof Node\Expr) {
+            return $this->getParametersForExpr($node->name, $node->getAttribute(AttributeKey::SCOPE));
+        }
+
+        return $this->getParametersForName($node->name, $node->getAttribute(AttributeKey::SCOPE));
+    }
+
+    /**
+     * @return array<int, ParameterReflection>
+     */
+    private function getParametersForExpr(Node\Expr $expr, Scope $scope): array
+    {
+        $type = $scope->getType($expr);
+
+        if (! $type instanceof ParametersAcceptor) {
+            return [];
+        }
+
+        return $type->getParameters();
+    }
+
+    /**
+     * @return array<int, ParameterReflection>
+     */
+    private function getParametersForName(Node\Name $name, Scope $scope): array
+    {
+        try {
+            return ParametersAcceptorSelector::selectSingle(
+                $this->broker->getFunction($name, $scope)->getVariants()
```

Comment:
Broker is very low-level class here. It should not be part of Rector rules.

What is your goal here of this method?

Reply:
Probably a lack of knowledge of phpstan internals on my side here.

Given a FuncCall identified by a Name, I want to return all Parameters.

Reply:
```php
$node->args
```

Reply:
No, not the args. The parameters defined in the signature of the function.

Reply:
In that case you can do simple `FunctionReflection`

Reply:
I must admit I'm a little lost.

Reply:
Keep you way than. This is just syntax suggar now